src/profile/ProfileForm.tsx

```diff
@@ -37,585 +25,387 @@ interface ChangedUser {
-            Role
+            Username
           </Form.Label>
           <Col sm={10}>
-            {/* NOTE: The custom attribute hasn't landed yet; it's here for future-compat. In the meantime, we add a
-              custom class instead. */}
-            <Form.Control
-              as="select"
-              name="is_editor"
-              onChange={onChange}
-              value={state.changedUser.is_editor?.toString()}
-              disabled={
-                auth.user?.username === user?.username && !user?.is_staff
+            <Field
+              errors={props.context.errors}
+              type="text"
+              name="username"
+              disabled={props.user !== undefined}
+              validMessage={"This username is valid."}
+              isValid={
+                props.context.formState.dirtyFields.has("username") &&
+                !props.context.errors["username"]
               }
-              className="custom-select"
-              custom
-            >
-              <option value="false">Contributor</option>
-              <option value="true">Editor</option>
-            </Form.Control>
+              onChange={async () => {
+                await props.context.triggerValidation("username");
+              }}
+              ref={props.context.register({
+                maxLength: {
+                  value: 150,
+                  message: "USER.USERNAME.TOO_LONG",
+                },
+                required: "USER.REQUIRED.username",
+                validate: async (username: string) => {
+                  if (props.user !== undefined) {
+                    // if we're editing a user the field is disabled anyway, don't validate
+                    return;
+                  }
+                  if (!(await validateUserNameDebounced(username))) {
+                    return Promise.resolve("USER.USERNAME.ALREADY_EXISTS");
+                  }
+                },
+              })}
+            />
           </Col>
         </Form.Group>
-      )}
-
-      <h3>Change password</h3>
-
-      {auth.user === user && (
-        <Form.Group as={Row} controlId="profileCurPassword">
+        <Form.Group as={Row}>
           <Form.Label column sm={2}>
-            Current password
+            Name
+          </Form.Label>
+          <Col sm={5}>
+            <Field
+              errors={props.context.errors}
+              type="text"
+              name="first_name"
+              placeholder="First Name"
+              ref={props.context.register({
+                maxLength: {
+                  value: 30,
+                  message: "USER.FIRST_NAME.TOO_LONG.30",
+                },
+                required: "USER.REQUIRED.first_name",
+              })}
+            />
+          </Col>
+          <Col sm={5}>
+            <Field
+              errors={props.context.errors}
+              type="text"
+              name="last_name"
+              placeholder="Last Name"
```

Comment:
Minor: since this is optional, the placeholder should mention that.

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/13/commits/c7182f99e122a2b497663f642b5ab8d7f63fa618